helm: deprecate metaMonitoring.grafanaAgent (#12307)
grafana/mimir: operations/helm/charts/mimir-distributed/CHANGELOG.md

## main / unreleased

> * [CHANGE] Grafana Agent: Depreate `metaMonitoring.grafanaAgent` values. #12307

Reviewer: ```suggestion
* [CHANGE] Grafana Agent: Deprecate `metaMonitoring.grafanaAgent` values. #12307
```

>   * Grafana Agent has been deprecated in earlier 2024, and will reach End-of-Support by the end of 2025.

Reviewer 2: ```suggestion
  * Grafana Agent was deprecated in early 2024 and reaches End-of-Support at the end of 2025.
```

>   * Instead of provisioning agent's Kubernetes resources with the chart's `metaMonitoring.grafanaAgent.enabled`, users are recommended to collect Mimir's meta-monitoring data with an external collector. We recommend Grafana k8s-monitoring, that manages the creation and lifecycle of Alloy instances, and has built-in support for collecting telemetry from Grafana LGTM stacks.

Reviewer 2: Do you mean `the agent's Kubernetes resources`? or, specifically, `Grafana Agent's Kubernetes resources`. 

It's best to be as specific as possible.

Author: Here I referred to "Kubernetes resources" (config maps, service monitors, access policies, etc), that our chart installs for Grafana Agent, when `metaMonitoring.grafanaAgent.enabled` is set. That is, the agent uses these resources to perform the meta-monitoring job.

Not sure how to phrase that better for the changelog.

Reviewer 2: I'd just add the article `the` to make it `Instead of provisioning the agent's . . . `.

>   * Instead of provisioning agent's Kubernetes resources with the chart's `metaMonitoring.grafanaAgent.enabled`, users are recommended to collect Mimir's meta-monitoring data with an external collector. We recommend Grafana k8s-monitoring, that manages the creation and lifecycle of Alloy instances, and has built-in support for collecting telemetry from Grafana LGTM stacks.

Reviewer 2: ```suggestion
  * Instead of provisioning agent's Kubernetes resources with the chart's `metaMonitoring.grafanaAgent.enabled`, collect Mimir's meta-monitoring data with an external collector. It's recommended to use Grafana k8s-monitoring, which manages the creation and lifecycle of Alloy instances and has built-in support for collecting telemetry from Grafana LGTM stacks.
```
